generator/internal/api/model.go

```diff
@@ -41,6 +46,56 @@ const (
 	SINT64_TYPE                 // 18
 )
 
+type Element interface {
+	Accept(v *Visitor) error
+}
+
+type Visitor interface {
+	VisitAPI(a *API) error
+	VisitMessage(m *Message) error
+	VisitService(s *Service) error
+	VisitMethod(m *Method) error
+	VisitMessageElement(m *MessageElement) error
+	VisitPathInfo(p *PathInfo) error
+	VisitPathSegment(s *PathSegment) error
+}
+
+// NoOpVisitor is a default Visitor implementation with no behavior.
+type NoOpVisitor struct{}
+
+func (n NoOpVisitor) VisitAPI(a *API) error {
+	return nil
+}
+
+func (n NoOpVisitor) VisitMessage(m *Message) error {
+	return nil
+}
+
+func (n NoOpVisitor) VisitService(s *Service) error {
+	return nil
+}
+
+func (n NoOpVisitor) VisitMethod(m *Method) error {
+	return nil
+}
+
+func (n NoOpVisitor) VisitMessageElement(m *MessageElement) error {
+	return nil
+}
+
+func (n NoOpVisitor) VisitPathInfo(p *PathInfo) error {
+	return nil
+}
+
+func (n NoOpVisitor) VisitPathSegment(s *PathSegment) error {
+	return nil
+}
+
+// InitializationVisitor is a Visitor used to initialize the API elements.
+type InitializationVisitor struct {
+	NoOpVisitor
+}
```

Comment:
Consider  refactoring `InitializationVisitor` to its own file. It may be easier to grok what it does if we put it all in one place.